benchmark/utils/DriverUtils.swift

```diff
@@ -167,28 +161,23 @@ struct TestConfig {
   }
 
   mutating func findTestsToRun() {
-    var i = 1
-    for benchName in precommitTests.keys.sorted() {
-      tests.append(Test(name: benchName, n: i, f: precommitTests[benchName]!))
-      i += 1
-    }
-    for benchName in otherTests.keys.sorted() {
-      tests.append(Test(name: benchName, n: i, f: otherTests[benchName]!))
-      i += 1
-    }
-    for benchName in stringTests.keys.sorted() {
-      tests.append(Test(name: benchName, n: i, f: stringTests[benchName]!))
-      i += 1
-    }
-    for i in 0..<tests.count {
-      if onlyPrecommit && precommitTests[tests[i].name] == nil {
-        tests[i].run = false
-      }
-      if !filters.isEmpty &&
-         !filters.contains(String(tests[i].index)) &&
-         !filters.contains(tests[i].name) {
-        tests[i].run = false
+    let allTests = [precommitTests, otherTests, stringTests]
+      .map { dictionary -> [(key: String, value: (Int)-> ())] in
+        Array(dictionary).sorted { $0.key < $1.key } } // by name
+      .joined()
+
+    let included: Set<String>
+      if !filters.isEmpty  { included = Set(filters) } else {
+        if onlyPrecommit { included = Set(precommitTests.keys) }
```

Comment:
I could give you what you ask for… but I've [obsessed about that particular detail a lot](https://gist.github.com/palimondo/25cfdc7902625cc7cf57d325ad898aee/revisions). Do you really think 
```swift
    let included: Set<String>
    if !filters.isEmpty  {
      included = Set(filters)
    } else {
      if onlyPrecommit {
        included = Set(precommitTests.keys)
      } else {
      included = Set(allTests.map { $0.key }) }
    }
```
is more clear way to express that we have 3 simple choices here? I don't think dogmatic adherence to one formatting style helps in **_this_** case.